vignettes/fileview.Rmd

```diff
@@ -0,0 +1,76 @@
+---
+title: "File View"
+author: "KimYen Ladia"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{File View}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## File View
+
+A file view is a view of all Files within one or more Projects or Folders. File views can:
+
+* Provide a way of isolating or linking data based on similarities
+* Provide the ability to link Files together by their annotations
+* Allow view/editing File attributes in bulk
+* Allow Files to be easily searched and queried
+
+Preliminaries:
+```{r collapse=TRUE}
+library(synapser)
+synLogin()
+# Create a new project
+projectName<-sprintf("My unique project created on %s", format(Sys.time(), "%a %b %d %H%M%OS4 %Y"))
+project<-Project(projectName)
+project<-synStore(project)
+
+# Create some files
+filePath<- tempfile()
+connection<-file(filePath)
+writeChar("this is the content of the first file", connection, eos=NULL)
+close(connection)  
+file <- File(path=filePath, parent=project)
+file <- synStore(file)
+
+filePath2<- tempfile()
+connection2<-file(filePath2)
+writeChar("this is the content of the second file", connection, eos=NULL)
+close(connection2)  
+file2 <- File(path=filePath2, parent=project)
+file2 <- synStore(file2)
+
+# add some annotations
+synSetAnnotations(file, list(contributor="Sage", class="V"))
+synSetAnnotations(file2, list(contributor="UW", rank="X"))
+```
+Creating a File View:
+
+```{r collapse=TRUE}
+# Notes: In python 1.8, we should be able to setup file view with annotation columns.
+view <- EntityViewSchema(name='my first file view',
+                         parent=project$properties$id,
+                         scopes=c(project$properties$id))
+
+view <- synStore(view)
+```
+To see the content of your newly created File View, use synTableQuery():
+```{r collapse=TRUE}
+query_results <- synTableQuery(sprintf("select * from %s", view$properties$id))
+data <- query_results$asDataFrame()
+data
+```
+## Updating Annotations using File View
```

Comment:
Since the schema using annotations cannot be set up programmatically, add instructions for doing so via the web client, then show how to write the query and update the annotations here. Once the Python client is extended we will update the vignette to be fully programmaticl